### ISSUE TITLE
Command 'reverse savereplay' randomly creates broken .omr files

### ISSUE THREAD

**Issue by Colpocorto** [openMSX/openMSX#1336]
* Version: OpenMSX 16.0 on Windows 7 X64.
* Machine emulated: Panasonic FS-A1GT, Panasonic FS-A1, Canon V-20 JP, others.
* Description:  when saving a replay by using command _reverse saveplay replayfile.omr_ the generated file cannot be loaded and the following message is shown:

_Fatal error: Cannot load replay, bad file format: myreplay.omr: Document parsing failed: unexpected end of data_

However, the issue happens randomly. Repeating the same sequence of steps several times in a row, some .omr files work fine while others do not.

The attachment includes some samples of both working and non-working .omr files.

[replays.zip](https://github.com/openMSX/openMSX/files/6214135/replays.zip)

**Comment by MBilderbeek** [comment]
Interesting!

I checked the diff between a non-working (rabocop.omr) and working (rabocop2.omr) file and saw that the former contains some illegal characters in the XML file (0x00 bytes).
In line 2857 the 's' of the filename 'srobocop.rom.SRAM' has been replaced by the 0x00 character (showing with an @ here):
```
                                <item version="2">
                                  <name>sramname</name>
                                  <data>@robocop.rom.SRAM</data>
                                  <attributes/>
                                  <children/>
                                </item>
```
exactly the same on line 6843. Apparently there are 2 snapshots in the file. I do wonder why there is an sramname serialized here. This seems to be part of a Konami typed megarom. (I also see it happening on my latest openMSX build.)

Anyway, when I remove the illegal character and replace it with 's' as it should be, it works fine. @m9710797 can you explain the sramname and could it be related to this bug?

**Comment by Colpocorto** [comment]
There is another game (kikikaikai.rom, standard ASCII8 mapper) that caused the same issue (see attachment).

BTW, as you talked about an XML and some other stuff, I guess that the .OMR file is a compressed bundle. Is there any standard way to extract the contents from it? It would be useful for diagnostic purposes.

[kikikaikai.omr.zip](https://github.com/openMSX/openMSX/files/6214355/kikikaikai.omr.zip)

**Comment by MBilderbeek** [comment]
It's a gzipped XML file. Just rename, gunzip and view XML.

Can you check whether you can also reproduce this issue (creating broken OMR files) with the latest development build?

**Comment by Colpocorto** [comment]
> It's a gzipped XML file. Just rename, gunzip and view XML.
> 
> Can you check whether you can also reproduce this issue (creating broken OMR files) with the latest development build?

I've just tried to reproduce the issue with the latest build available for Windows X64 (2021-03-24) and after creating 20 .omr files, all of them seemed to work fine.  I used different setups (msx1, msx2, MSXTurboR) and everything worked as intended.  However, being this issue a bit 'random', I think that further testing should be made.

By the way, I've detected another possible bug in this build: if a non-existing machine is configured with the '-machine' switch, nothing happens (the emulator won't boot, but no error message is shown). Shall I create another issue for that, or is it something to be expected?

Thank you.

**Comment by MBilderbeek** [comment]
Thanks for testing that. Please keep testing for a bit. How easy do you get it with openMSX 16.0?

The other thing is also not expected. Can you make a new issue for it? This is what happens here:
```
$ openmsx -machine bla
Fatal error: Error in "bla" machine: machines/bla.xml not found in this context
```

**Comment by Colpocorto** [comment]
I find a broken .omr about 25% of the attempts to save a replay with official openMSX 16.0.
I will keep testing a bit more.

Thank you for your support.

**Comment by m9710797** [comment]
Hi,
I can confirm that this issue is already fixed.
It was fixed on Jan 1 2021 in f1e5d84847e7320b634d0df03f37ac5d06847e17.
I believe this bug was introduced on Nov 14 202 in 476e352ffe52e41ebe3a916dbf283fb0ba6cc481 (though not 100% sure about this one).
@Colpocorto But thanks for reporting anyway!
Wouter

**Comment by Colpocorto** [comment]
Thank you.
